src/fluree/db/api/ledger.cljc

```diff
@@ -28,37 +31,126 @@
      ;; session/db returns a promise channel
      (session/db conn ledger nil))))
 
+(defn- add-db-auth-sid
+  "Resolves auth subject id from any identity value. Will
+  throw an exception if it is unable to resolve to an established identity.
+  Auth 0 is a special case. It:
+  - Is short-hand to keeping 'local' root-level permissions. 'local' permissions
+    may already be a restricted permissioned set of data, it simply is not further restricted.
+  - Does not attempt to resolve to a subject ID (as it would error as '0' will not exist in the db)"
+  [db auth]
+  (go-try
+    (cond
+      (not auth)
+      db
+
+      (= 0 auth)
+      (assoc db :auth 0)
+
+      :else
+      (let [auth-sid (<? (dbproto/-subid db auth))]
+        (if auth-sid
+          (assoc db :auth auth-sid)
+          (throw (ex-info (str "Auth id: " auth " unknown.")
+                          {:status 401
+                           :error  :db/invalid-auth})))))))
+
+
+(defn- add-db-permissions
+  "Adds permissions to db. Permissions can either be explicitly stated with roles
+  or it can be derived from an auth-id.
+  This assumes the :auth on the db, if it was provided, is already resolved
+  via add-db-auth-sid function."
+  [db auth roles]
+  (go-try
+    (let [auth-sid    (cond
+                        (nil? auth) nil
+                        (= 0 auth) 0
+                        auth (or (<? (dbproto/-subid db auth))
+                                 (throw (ex-info (str "Auth id: " auth " unknown.")
+                                                 {:status 401
+                                                  :error  :db/invalid-auth}))))
+          roles'      (cond
+                        roles roles
+                        auth-sid (<? (auth/roles db auth-sid))
+                        :else nil)
+          permissions (when roles'
+                        (<? (permissions/permission-map db roles' :query)))]
+      (assoc db :auth auth-sid
+                :roles roles'
+                :permissions permissions))))
+
+
+(defn- syncTo-wait
```

Comment:
I would prefer to see Clojure idiomatic kebab-case naming in the code even if the API uses camelCase. So `sync-to-wait`, etc.

Reply:
This is there because the option is {:opts {:syncTo 1234}} - to make it specific this function handles the `syncTo` option.

Reply:
Yeah, I figured so. But unless the code needs to do something like "convert the keyword to a symbol and call it as a fn" (which might have some security implications), I still think it would be preferable to use Clojure idioms for naming symbols and other identifiers in the code. Small thing but it adds up when these things are consistent and predictable across multiple codebases and multiple developers. It's also likely going to be a thing external contributors notice and comment on.